parser.y

```diff
@@ -1738,26 +1738,24 @@ AlgorithmClause:
 	}
 
 LockClause:
-	"LOCK" EqOpt "NONE"
-	{
-		$$ = ast.LockTypeNone
-	}
-|	"LOCK" EqOpt "DEFAULT"
+	"LOCK" EqOpt "DEFAULT"
 	{
 		$$ = ast.LockTypeDefault
 	}
-|	"LOCK" EqOpt "SHARED"
-	{
-		$$ = ast.LockTypeShared
-	}
-|	"LOCK" EqOpt "EXCLUSIVE"
-	{
-		$$ = ast.LockTypeExclusive
-	}
-|	"LOCK" EqOpt identifier
+|	"LOCK" EqOpt Identifier
 	{
-		yylex.AppendError(ErrUnknownAlterLock.GenWithStackByArgs($3))
-		return 1
+		id := strings.ToUpper($3)
+
+		if id == "NONE" {
```

Comment:
how about handle `Default` here?

Reply:
```go
	"LOCK" EqOpt "DEFAULT"
	{
		$$ = ast.LockTypeDefault
	}
```
See https://github.com/pingcap/parser/pull/516/files#diff-966f3247252fb3ffd7810a3b943884f0R1741

Reply:
I mean that we remove `"LOCK" EqOpt "DEFAULT"` rule and add a branch `if id == "DEFAULT"` in line 1749

Reply:
Oh, I misread it.
`"DEFAULT"` is reserved keyword, it's not included in `Identifier`.
```go
Identifier:
identifier | UnReservedKeyword | NotKeywordToken | TiDBKeyword
```

Reply:
fine LGTM